pkg/registry/registry_test.go

```diff
@@ -87,6 +88,153 @@ OdoSettings:
+					"icon": "/images/angular.svg",
+					"globalMemoryLimit": "2686Mi",
+					"links": {
+						"self": "/devfiles/angular/devfile.yaml"
+					}
+				}
+			]
+			`,
+		))
+		if err != nil {
+			t.Error(err)
+		}
+	}))
+	// Close the server when test finishes
+	defer server.Close()
+
+	// Create a temporary file with the above server URL
+	tempConfigFile, err := ioutil.TempFile("", "odoconfig")
+	if err != nil {
+		t.Fatal("Fail to create temporary config file")
+	}
+	defer os.Remove(tempConfigFile.Name())
+	defer tempConfigFile.Close()
+	_, err = tempConfigFile.Write([]byte(
+		`kind: Preference
+apiversion: odo.openshift.io/v1alpha1
+OdoSettings:
+  RegistryList:
+  - Name: TestRegistry
+    URL: ` + server.URL,
+	))
+	if err != nil {
+		t.Error(err)
+	}
+
+	os.Setenv(preference.GlobalConfigEnvName, tempConfigFile.Name())
+	defer os.Unsetenv(preference.GlobalConfigEnvName)
+
+	const registryName = "TestRegistry"
+	tests := []struct {
+		name         string
+		registryName string
+		want         DevfileStackList
+	}{
+		{
+			name:         "Case 1: Test getting ALL registries and looking for nodejs",
+			registryName: "",
+			want: DevfileStackList{
+				DevfileRegistries: []Registry{
+					{
+						Name:   "TestRegistry",
+						URL:    server.URL,
+						Secure: false,
+					},
+				},
+				Items: []DevfileStack{
+					{
+						Name:        "nodejs",
+						DisplayName: "NodeJS Angular Web Application",
+						Description: "Stack for developing NodeJS Angular Web Application",
+						Registry: Registry{
+							Name: registryName,
+							URL:  server.URL,
+						},
+						Link:     "/devfiles/angular/devfile.yaml",
+						Language: "nodejs",
+						Tags:     []string{"NodeJS", "Angular", "Alpine"},
+					},
+				},
+			},
+		},
+		{
+			name:         "Case 2: Test getting from only one specific registry",
+			registryName: "TestRegistry",
+			want: DevfileStackList{
+				DevfileRegistries: []Registry{
+					{
+						Name:   "TestRegistry",
+						URL:    server.URL,
+						Secure: false,
+					},
+				},
+				Items: []DevfileStack{
+					{
+						Name:        "nodejs",
+						DisplayName: "NodeJS Angular Web Application",
+						Description: "Stack for developing NodeJS Angular Web Application",
+						Registry: Registry{
+							Name: registryName,
+							URL:  server.URL,
+						},
+						Link:     "/devfiles/angular/devfile.yaml",
+						Language: "nodejs",
+						Tags:     []string{"NodeJS", "Angular", "Alpine"},
+					},
+				},
+			},
+		},
+		{
+			name:         "Case 3: Expect nothing back if registry is not found",
+			registryName: "Foobar",
+			want: DevfileStackList{
+				// We use "nil" here as reflect.DeepEqual() returns false if one slice is nil,
+				// and the other is a non-nil slice with 0 length.
+				// So we simply just say 'nil'
+				DevfileRegistries: nil,
+				Items:             nil,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			prefClient, _ := preference.NewClient()
```

Comment:
Can you use Mock client instead? See https://github.com/redhat-developer/odo/blob/main/pkg/init/init_test.go#L65 for reference.

Reply:
Didn't need to since we're not modifying the clients (clients are both "blank"), so it was cleaner to implement it this way, instead of via fields.

What do you think @feloy ?

Reply:
By mocking ths Preference client, you could mock the methods on this client the tested function is using only, without needing to create the underlying file. This way, you should not rely on the implementation of the preference package

Reply:
@feloy Unfortunately https://github.com/redhat-developer/odo/blob/main/pkg/registry/registry.go#L175 has way too many HTTP requests and because it's not part of the preference or registry client, I'd be unable to mock it without refactoring that side of code (which is outside the scope of this PR).

Could we keep these tests (which are based on the `odo dev` unit tests) and revisit it at a later date to refactor?

At the moment, they pass, they work well.

Reply:
@cdrage I don't think it is related. You can see on this commit how to mock the preference client to avoid using a preference file:
https://github.com/feloy/odo/commit/6551ebe2d74281b87b0a4b50f76ba78168da1713

Reply:
Ohhhh! I thought you were asking me to also somehow mock the http server... which would of been a lot.

I'll add the preference mock